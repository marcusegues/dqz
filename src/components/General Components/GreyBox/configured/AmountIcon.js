// @flow
import React from 'react';
// $FlowFixMe
import { Text, StyleSheet } from 'react-native';
import { GreyBox } from '../GreyBox';

const ownStyles = StyleSheet.create({
  amountText: {
    color: 'black',
    fontSize: 14,
    fontFamily: 'roboto_regular',
  },
});

type AmountIconProps = {
  amount: number,
  currency: string,
};

export const AmountIcon = ({ amount, currency }: AmountIconProps) => (
  <GreyBox>
    <Text style={ownStyles.quantityText}>{`${currency} ${amount.toFixed(
      2
    )}`}</Text>
  </GreyBox>
);
