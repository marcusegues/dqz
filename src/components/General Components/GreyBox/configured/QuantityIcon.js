// @flow
import React from 'react';
// $FlowFixMe
import { Text, StyleSheet } from 'react-native';
import { GreyBox } from '../GreyBox';

const ownStyles = StyleSheet.create({
  quantityText: {
    color: 'black',
    fontSize: 14,
    fontFamily: 'roboto_regular',
  },
});

type QuantityIconProps = {
  quantity: number | string,
  unit: string,
};

export const QuantityIcon = ({ quantity, unit }: QuantityIconProps) => (
  <GreyBox>
    <Text style={ownStyles.quantityText}>{`${quantity} ${unit}`}</Text>
  </GreyBox>
);
