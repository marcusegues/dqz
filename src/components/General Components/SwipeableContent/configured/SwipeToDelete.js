// @flow
import React from 'react';
// $FlowFixMe
import { TouchableHighlight } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { SwipeableContent } from '../SwipeableContent';
import type { Children } from '../../../../types/generalTypes';

type SwipeToDeleteProps = {
  onPressDelete: Function,
  children: Children,
};

export const SwipeToDelete = ({
  onPressDelete,
  children,
}: SwipeToDeleteProps) => {
  const rightSwipeButtons = [
    <TouchableHighlight
      style={{
        flex: 1,
        backgroundColor: 'rgb(217,10,35)',
        justifyContent: 'center',
      }}
      onPress={onPressDelete}
    >
      <Ionicons
        name="md-trash"
        size={30}
        style={{ color: 'white', paddingLeft: 35 }}
      />
    </TouchableHighlight>,
  ];

  return (
    <SwipeableContent
      rightSwipeButtons={rightSwipeButtons}
      rightButtonWidth={90}
    >
      {children}
    </SwipeableContent>
  );
};
