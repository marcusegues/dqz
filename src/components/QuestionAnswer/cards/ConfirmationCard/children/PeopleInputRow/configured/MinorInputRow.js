import React from 'react';
import { View, Text } from 'react-native';
import { translate } from 'react-i18next';
import { PeopleInputRow } from '../PeopleInputRow';
import * as fromModelApi from '../../../../../../../model/configurationApi';
import { moderateScale } from '../../../../../../../styles/Scaling';

export const ownStyles = {
  categoryTitleText: {
    fontFamily: 'roboto_medium',
    fontSize: moderateScale(13),
    color: '#24253D',
  },
  categoryAgeText: {
    fontFamily: 'roboto_regular',
    fontSize: moderateScale(11),
    color: '#898989',
  },
};

const kids = require('../../../../../../../../assets/icons/kids.png');

const MinorInputRow = ({ onAddMinor, onSubtractMinor, people, t }) => (
  <PeopleInputRow
    onAdd={onAddMinor}
    onSubtract={onSubtractMinor}
    people={people}
    icon={kids}
    displayedQuantity={fromModelApi.getMinorPeople(people)}
  >
    {{
      textComponent: (
        <View>
          <Text style={ownStyles.categoryTitleText}>{t('minor')}</Text>
          <Text style={ownStyles.categoryAgeText}>{t('minorInfo')}</Text>
        </View>
      ),
    }}
  </PeopleInputRow>
);

export default translate(['qaFlow'])(MinorInputRow);
