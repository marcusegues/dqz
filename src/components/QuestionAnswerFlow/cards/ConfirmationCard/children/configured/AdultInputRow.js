import React from 'react';
import { View, Text } from 'react-native';
import PeopleInputRow from '../PeopleInputRow';
import * as fromModelApi from '../../../../../../model/configurationApi';
import { moderateScale } from '../../../../../../styles/Scaling';

const AdultInputRow = ({ onAddAdult, onSubtractAdult, people }) => (
  <PeopleInputRow
    onAdd={onAddAdult}
    onSubtract={onSubtractAdult}
    people={people}
    icon={`human-male`}
    displayedQuantity={fromModelApi.getAdultPeople(people)}
  >
    {{
      textComponent: (
        <View>
          <Text
            style={{
              fontFamily: 'roboto_medium',
              fontSize: moderateScale(14),
              color: '#24253D',
            }}
          >{`Erwachsene`}</Text>
          <Text
            style={{
              fontFamily: 'roboto_regular',
              fontSize: moderateScale(12),
              color: '#898989',
            }}
          >{`17+ Jahre`}</Text>
        </View>
      ),
    }}
  </PeopleInputRow>
);

export default AdultInputRow;
