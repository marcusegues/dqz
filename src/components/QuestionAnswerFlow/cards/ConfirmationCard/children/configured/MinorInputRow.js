import React from 'react';
import { View, Text } from 'react-native';
import PeopleInputRow from '../PeopleInputRow';
import * as fromModelApi from '../../../../../../model/configurationApi';
import { moderateScale } from '../../../../../../styles/Scaling';

const MinorInputRow = ({ onAddMinor, onSubtractMinor, people }) => (
  <PeopleInputRow
    onAdd={onAddMinor}
    onSubtract={onSubtractMinor}
    people={people}
    icon={`human-child`}
    displayedQuantity={fromModelApi.getMinorPeople(people)}
  >
    {{
      textComponent: (
        <View>
          <Text
            style={{
              fontFamily: 'roboto_medium',
              fontSize: moderateScale(14),
              color: '#24253D',
            }}
          >{`Kinder & Jugendliche`}</Text>
          <Text
            style={{
              fontFamily: 'roboto_regular',
              fontSize: moderateScale(12),
              color: '#898989',
            }}
          >{`unter 17 Jahre`}</Text>
        </View>
      ),
    }}
  </PeopleInputRow>
);

export default MinorInputRow;
