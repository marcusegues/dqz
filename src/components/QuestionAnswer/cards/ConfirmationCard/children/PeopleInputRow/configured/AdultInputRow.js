// @flow
import React from 'react';
import type { ComponentType } from 'react';
// $FlowFixMe
import { View, Text } from 'react-native';
import { translate } from 'react-i18next';
import { PeopleInputRow } from '../PeopleInputRow';
import * as fromModelApi from '../../../../../../../model/configurationApi';

import { ownStyles } from './MinorInputRow';
import type { People } from '../../../../../../../model/types/basketPeopleAmountsTypes';
import type { TFunction } from '../../../../../../../types/generalTypes';

const adults = require('../../../../../../../../assets/icons/adult.png');

type AdultInputRowProps = {
  onAddAdult: () => void,
  onSubtractAdult: () => void,
  people: People,
};

const AdultInputRowInner = ({
  onAddAdult,
  onSubtractAdult,
  people,
  t,
}: AdultInputRowProps & { t: TFunction }) => (
  <PeopleInputRow
    onAdd={onAddAdult}
    onSubtract={onSubtractAdult}
    people={people}
    icon={adults}
    displayedQuantity={fromModelApi.getAdultPeople(people)}
  >
    <View>
      <Text style={ownStyles.categoryTitleText}>{t('adult')}</Text>
      <Text style={ownStyles.categoryAgeText}>{t('adultInfo')}</Text>
    </View>
  </PeopleInputRow>
);

export const AdultInputRow = (translate(['qaFlow'])(
  AdultInputRowInner
): ComponentType<AdultInputRowProps>);
