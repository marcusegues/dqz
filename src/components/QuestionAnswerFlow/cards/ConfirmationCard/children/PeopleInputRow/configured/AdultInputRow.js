import React from 'react';
import { View, Text } from 'react-native';
import PeopleInputRow from '../PeopleInputRow';
import * as fromModelApi from '../../../../../../../model/configurationApi';

import { ownStyles } from './MinorInputRow';

const AdultInputRow = ({ onAddAdult, onSubtractAdult, people }) => (
  <PeopleInputRow
    onAdd={onAddAdult}
    onSubtract={onSubtractAdult}
    people={people}
    icon="human-male"
    displayedQuantity={fromModelApi.getAdultPeople(people)}
  >
    {{
      textComponent: (
        <View>
          <Text style={ownStyles.categoryTitleText}>Erwachsene</Text>
          <Text style={ownStyles.categoryAgeText}>17+ Jahre</Text>
        </View>
      ),
    }}
  </PeopleInputRow>
);

export default AdultInputRow;
