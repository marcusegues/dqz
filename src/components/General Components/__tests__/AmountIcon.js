// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { AmountIcon } from '../../General Components/GreyBox/configured/AmountIcon';

describe('AmountIcon TestSuite', () => {
  test('AmountIcon component renders correctly', () => {
    const tree = renderer
      .create(<AmountIcon currency="EUR" amount={22} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
