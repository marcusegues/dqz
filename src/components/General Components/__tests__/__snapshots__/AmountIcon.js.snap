// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountIcon TestSuite AmountIcon component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E0E0E1",
      "borderRadius": 3,
      "paddingHorizontal": 8,
      "paddingVertical": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    EUR 22.00
  </Text>
</View>
`;
